openfe/analysis/plotting.py

```diff
@@ -32,7 +38,25 @@ def plot_lambda_transition_matrix(matrix: npt.NDArray) -> Axes:
             ax.axhline(y=i, ls="-", lw=0.5, color="k", alpha=0.25)
         for j in range(num_states):
             val = matrix[i, j]
-            val_str = "{:.2f}".format(val)[1:]
+
+            # Catch if 0.05 from 0 or 1
+            # https://github.com/OpenFreeEnergy/openfe/issues/806
+            if matrix[j, i] < 0.005:
+                # This replicates the same behaviour as alchemical-analysis & alchemlyb
+                # i.e. near-zero values will just not be annotated
+                val_str = ""
+            elif matrix[j, i] > 0.995:
+                val_str = "{:.2f}".format(matrix[j, i])[:4]
+                # Throw a warning if the value is greater than float error
+                # This means your overlap matrix is probably wrong
+                if matrix[j, i] > 1.01:
+                    wmsg = (f"Overlap probability exceeds 1.0: {matrix[j, i]} "
+                            "This is inddicates an incorrect overlap matrix "
```

Comment:
```suggestion
                            "This indicates an incorrect overlap matrix "
```

Reply:
For the first bad overlap matrix (the one that goes up to 155), the first few rows only sum up to 50 something. Should we also add a warning when rows generally don't add up to ~100 (so either far below or above that) or do you think that such a case would always also have a value in the matrix that is >1.01?

Reply:
That's a good question - it might be better to just sum up each row & column and check if any of them exceeds 1.01? That's the same as checking if any individual cell exceeds > 1.0.1

Reply:
Updated, please let me know what you think @hannahbaumann - wasn't too sure what to call the thing since it takes both a replica exchange probability matrix and an mbar state overlap probability matrix.